src/main/kotlin/sidechain/eth/EthChainHandler.kt

```diff
@@ -142,11 +147,19 @@ class EthChainHandler(
                 val time = block.block.timestamp.multiply(BigInteger.valueOf(1000))
                 block.block.transactions
                     .map { it.get() as Transaction }
-                    .flatMap {
-                        if (wallets.containsKey(it.to))
-                            handleEther(it, time, wallets)
-                        else if (tokens.containsKey(it.to))
-                            handleErc20(it, time, wallets, tokens)
+                    .flatMap { tx ->
+                        val receipt = web3.traceTransaction(tx.hash).send()
+                        receipt.result.forEach {
+                            val ac = it.action
+                            if (ac is Trace.CallAction) {
+                                if (wallets.containsKey(ac.to)) {
```

Comment:
can we do this? ` if (ac is Trace.CallAction && wallets.containsKey(ac.to))` it looks more clear to me.